src/concerns/dimensions.coffee

```diff
@@ -7,14 +7,20 @@ export default
 		viewportWidth: null # Width of the viewport, for media query calculation
 		carouselWidth: null # Width of a page of the carousel
 		gutterWidth: 0 # Computed width of gutters, since they support css vars
+		resizeObserver: null
 
 	# Add resize listening
 	mounted: ->
 		@onResize()
-		window.addEventListener 'resize', @onResize
 
-	# Cleanup listeners
-	beforeDestroy: -> window.removeEventListener 'resize', @onResize
+		# resize observer listens for the element itself
+		# changing dimensions
+		@resizeObserver = new ResizeObserver @onResize
+		@resizeObserver.observe @$el
+
+	beforeDestroy: ->
+		if @$el then @resizeObserver.unobserve(@$el)
+		else @resizeObserver.disconnect()
```

Comment:
Should we just do `@resizeObserver.disconnect()`?  Since docs say

> The disconnect() method of the [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) interface unobserves all observed [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element) or [SVGElement](https://developer.mozilla.org/en-US/docs/Web/API/SVGElement) targets.

```suggestion
		@resizeObserver?.disconnect()
```

Reply:
When I was reading about how to use it, the article suggested doing both as a percaution, but I understand